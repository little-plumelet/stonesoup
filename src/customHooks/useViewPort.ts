import { useEffect, useState } from 'react';

export const useVeiwPort = () => {
  const [width, setWidth] = useState(window.innerWidth);

  useEffect(() => {
    function resizeWidth() {
      setWidth(window.innerWidth);
    }
    window.addEventListener('resize', resizeWidth);
    return () => {
      window.removeEventListener('resize', resizeWidth);
    };
  }, []);

  return { width };
};
