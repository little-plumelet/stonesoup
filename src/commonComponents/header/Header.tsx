import { SearchOutlined, UserOutlined } from '@ant-design/icons';
import { Button, Space } from 'antd';
import { useLocation, useNavigate } from 'react-router-dom';
import style from './style.module.css';
import { NavBar } from './parts/navBar';
import { SearchBar } from './parts/searchBar';
import logo from '../../assets/pot_logo.png';
import { MobileNavBar } from './parts/mobileNavBar';
import { useVeiwPort } from '../../customHooks/useViewPort';

export function Header() {
  const location = useLocation();
  const navigate = useNavigate();
  const { width } = useVeiwPort();
  const breackPoint = 620;

  const handleSearchClick = () => {
    navigate('/searchresult');
  };

  return (
    <header className={style.headerBlock}>
      <div className={style.header}>
        <div className={style.logoBlock}>
          <img className={style.logoImg} src={logo} alt="logo" />
          <h1 className={style.logo}>StoneSoup</h1>
          <h1 className={style.logoClone}>StoneSoup</h1>
        </div>
        {width >= breackPoint && (
          <NavBar
            navStyle={{
              display: 'flex',
              justifyContent: 'space-around',
              gap: '1rem',
            }}
          />
        )}
        <Space size="middle">
          <Button
            shape="circle"
            icon={<SearchOutlined />}
            onClick={handleSearchClick}
            data-testid="searchicon"
          />
          <Button shape="circle" icon={<UserOutlined />} />
          {width < breackPoint && <MobileNavBar />}
        </Space>
      </div>
      <div className={style.searchBar}>
        {location?.pathname === '/searchresult' && <SearchBar />}
      </div>
    </header>
  );
}
