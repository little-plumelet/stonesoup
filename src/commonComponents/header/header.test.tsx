import { fireEvent, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { MemoryRouter, Router } from 'react-router-dom';
import * as reactHooks from 'react-redux';
import { Header } from './Header';
import { useAppDispatch, useAppSelector } from '../../customHooks/hooksRedux';

jest.mock('react-redux');

const dispatch = jest.fn();
const foo = { useAppSelector, useAppDispatch };

jest.spyOn(foo, 'useAppDispatch').mockReturnValue(dispatch);
jest.spyOn(reactHooks, 'useDispatch').mockReturnValue(dispatch);
jest.spyOn(reactHooks, 'useDispatch').mockReturnValue(dispatch);

describe('header', () => {
  it('should render', () => {
    const component = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });

  it('handle searchIcon click', () => {
    const history = createMemoryHistory();
    render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    fireEvent.click(screen.getByTestId('searchicon'));
    expect(history.location.pathname).toBe('/searchresult');
  });

  it('check searchBar', () => {
    const history = createMemoryHistory({ initialEntries: ['/searchresult'] });
    render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    expect(history.location.pathname).toBe('/searchresult');
    expect(screen.getByTestId('searchbar')).toBeInTheDocument();
  });
});
