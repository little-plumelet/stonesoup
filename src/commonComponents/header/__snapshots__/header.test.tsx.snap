// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="headerBlock"
      >
        <div
          class="header"
        >
          <a
            href="/"
          >
            <div
              class="logoBlock"
            >
              <img
                alt="logo"
                class="logoImg"
                src=""
              />
              <h1
                class="logo"
              >
                StoneSoup
              </h1>
              <h1
                class="logoClone"
              >
                StoneSoup
              </h1>
            </div>
          </a>
          <nav
            id="nav"
            style="display: flex; justify-content: space-around; gap: 1rem;"
          >
            <a
              aria-current="page"
              class="navLink, active"
              href="/"
            >
              Home
            </a>
            <a
              class="navLink, pending"
              href="/about"
            >
              About Us
            </a>
          </nav>
          <div
            class="ant-space css-dev-only-do-not-override-mxhywb ant-space-horizontal ant-space-align-center icons"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
                data-testid="searchicon"
                type="button"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
                type="button"
              >
                <span
                  aria-label="user"
                  class="anticon anticon-user"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="searchBar"
        />
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="headerBlock"
    >
      <div
        class="header"
      >
        <a
          href="/"
        >
          <div
            class="logoBlock"
          >
            <img
              alt="logo"
              class="logoImg"
              src=""
            />
            <h1
              class="logo"
            >
              StoneSoup
            </h1>
            <h1
              class="logoClone"
            >
              StoneSoup
            </h1>
          </div>
        </a>
        <nav
          id="nav"
          style="display: flex; justify-content: space-around; gap: 1rem;"
        >
          <a
            aria-current="page"
            class="navLink, active"
            href="/"
          >
            Home
          </a>
          <a
            class="navLink, pending"
            href="/about"
          >
            About Us
          </a>
        </nav>
        <div
          class="ant-space css-dev-only-do-not-override-mxhywb ant-space-horizontal ant-space-align-center icons"
        >
          <div
            class="ant-space-item"
            style="margin-right: 16px;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
              data-testid="searchicon"
              type="button"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-circle ant-btn-default ant-btn-icon-only"
              type="button"
            >
              <span
                aria-label="user"
                class="anticon anticon-user"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="searchBar"
      />
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
